<!-- usage: roslaunch ssc_planning launch_fusion_evaluation.launch fusion_strategy:=sc_fusion
 eval_directory:=/Data/simulations/Data/ExperimentConfiguration/ssc_fusion_evaluation/results/sc_fusion -->

<launch>
  <!-- Voxblox arguments -->
  <arg name="voxel_size" default="0.08"/>
  <arg name="voxels_per_side" default="16"/>
  <arg name="ray_length" default="5.0"/>
  <arg name="method" default="merged"/>
  <arg name="verbose_voxblox" default="false"/>
  <arg name="fusion_strategy" default="naive"/>
  <arg name="eval_directory" default="/Data/simulations/Data"/>
  <param name="/use_sim_time" value="true"/>
 
  
<!--     Voxblox -->
 <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server"   output="screen" args="-alsologtostderr">
   <remap from="pointcloud" to="/airsim_drone/RGB_pcl"/>
   <param name="use_tf_transforms" value="true"/>
   <param name="publish_tsdf_map" value="true" />
   <param name="publish_pointclouds_on_update" value="true" />
   <param name="publish_esdf_map" value="true" />
   <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
   <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
   <param name="color_mode" value="color" />
   <param name="update_mesh_every_n_sec" value="1.0" />
   <param name="method" value="$(arg method)" />
   <param name="max_ray_length_m" value="25" />
   <param name="allow_clear" value="true" />
   <param name="verbose" value="$(arg verbose_voxblox)" />
   <param name="publish_map_incremental" value="true" />
 </node>

  <node name="ssc_node" pkg="ssc_mapping" type="ssc_mapping_node"  required="true" args="-alsologtostderr"  output="screen">
   <param name="publish_pointclouds" value="true" />
   <param name="ssc_topic" value="ssc" />
   <param name="ssc_voxel_size" value="$(arg voxel_size)" />
   <param name="ssc_voxels_per_side" value="$(arg voxels_per_side)" />
   <param name="fusion_strategy" value="$(arg fusion_strategy)" />
   <param name="fusion_prob_free" value="0.475" />  
   <param name="fusion_prob_occupied" value="0.6775" />  
   <param name="fusion_max_weight" value="30.0" />   
   <param name="fusion_pred_conf" value="0.75" />   
   <param name="fusion_min_prob" value="0.12" />  
   <param name="fusion_max_prob" value="0.97" /> 
 </node>

 <node pkg="rosbag" type="play" name="player" required="true" output="screen" args=" --rate 1.5 --clock /Data/simulations/Data/subset_ssc_prob.bag"/>

 <node name="eval_data_node" pkg="active_3d_planning_app_reconstruction" type="eval_data_node.py" output="screen" required="true">
      <remap from="ue_out_in" to="/airsim_drone/RGB_pcl" />
      <remap from="collision" to="/airsim_drone/collision" />
      <param name="ns_planner" value="/planner_node" />
      <param name="ns_voxblox" value="/voxblox_node" />
      <param name="start_planner" value="false" />
      <param name="ns_ssc" value="/ssc_node" />
      <param name="delay" value="3.0" />
      <param name="startup_timeout" value="30.0" />
      <param name="evaluate" value="true" />
      <param name="eval_directory" value="$(arg eval_directory)" />
      <param name="gt_map_path" value="/home/mansoor/flat_filled_free.tsdf" />
      <param name="eval_frequency" value="30" />
      <param name="time_limit" value="15" />
    </node>


  <!-- RVIZ Visualization -->
  <node type="rviz" name="rviz_voxblox" pkg="rviz" args="-d $(find ssc_planning)/cfg/visualization/visualization.rviz"/>
  
</launch>
